### DISCUSSION TITLE
Tutorial on using create-block scaffolding to write a multi-block plugin

### DISCUSSION THREAD

**Discussion by bph** [WordPress/developer-blog-content#186] [Topic Ideas]
The first impression is that create-block scaffolding creates only single block plugins and that it can be used to add multiple blocks to a plugin. 

It could be a block that requires additional blocks as inner blocks/ And example could be a one question poll. with multiple answers.

**Comment by bph** [comment]
We'll keep the discussion open for a while longer, until we can assign it to a writer.

**Comment by n8finch** [comment]
Hey @bph is there a standard way to do this via `create-block`? I couldn't find an issue or flag in Gutenberg repo. I'd be happy to write the tutorial 🙂

What I usually do:
1.  is run `create-block` to scaffold out the plugin
2. delete what's in `src`
3. `cd` into `src`, run `create-block --no-plugin`
4. add the blocks to the `init` hook via `register_block_type()`, adjust script build directory
5. Make sure that the scripts are building to the build directory.

So the structure looks like:
```
block-plugin
|--src
|--\block-1
|--|--\--block files like edit.js, render.php, etc.
|--\block-2
|--|--\--block files like edit.js, render.php, etc.
|--build
|--\block-1
|--|--\ --built files
|--\block-2
|--|--\ --built files
|--package.json
|--block-plugin.php
```

Something like that.

Let me know if there's better directives on this 🙂

**Reply by justintadlock** [reply]
CC'ing @ryanwelcher here. I know he's done it, but I forgot how it works.

**Reply by n8finch** [reply]
This is definitely a good starting point (from Ryan's YouTube): https://www.youtube.com/watch?v=kPXoT-GOusY

But doesn't use `create-block`, @ryanwelcher definitely curious on your feedback. Let me know if you need more than the above "psuedo-directory". 🙂

**Reply by ryanwelcher** [reply]
I think this is a great idea. I built a template a while ago that does this but it's probably extremely out of date now - https://www.npmjs.com/package/@ryanwelcher/multiple-blocks-template - you're welcome to use that as reference if it's helpful.

@n8finch  As for the directory structure, I think that is completely up to you and shows the power of the templating system that create-block uses. Let me know if I can be of any help!

**Reply by n8finch** [reply]
Thanks for the link @ryanwelcher, I appreciate it.

@bph let me know if you'd like me to move forward with this, or if you need anything else, or would like someone else to work on this (no worries if so 🙂)

**Reply by justintadlock** [reply]
I can set it up for you. Give me a few, and I'll assign a ticket for you.

**Reply by justintadlock** [reply]
@n8finch - If you'll just comment on this ticket, I can assign it to you: https://github.com/WordPress/developer-blog-content/issues/204

It's a weird thing where we can't assign tickets until someone has commented. 🤷

**Comment by bph** [comment]
Thank you, @n8finch for taking this on. Looking forward to reading your post!

**Comment by bph** [comment]
Post published: [Setting up a multi-block plugin using InnerBlocks and post meta](https://developer.wordpress.org/news/2024/05/22/setting-up-a-multi-block-using-inner-blocks-and-post-meta/)
